File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {

        size_t l {0};
        size_t r {numbers.size() - 1};

        while (l < r){
            int cur {numbers[l] + numbers[r]};

            if (cur == target) {
                return {static_cast<int>(l + 1), static_cast<int>(r + 1)};
            } else if (cur > target){
                r--;
            } else {
                l++;
            }
        }

        return {};
    }
};

File: Data Structures & Algorithms/target-sum/submission-1.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:

        res = 0

        def backtrack(i, cur):
            if i == len(nums):
                if cur == target:
                    nonlocal res
                    res += 1
                return

            backtrack(i + 1, cur + nums[i])
            backtrack(i + 1, cur - nums[i])

        backtrack(0, 0)
        return res

        